runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -315,10 +307,23 @@ private Batcher() {
       this.queue = new HashMap<>();
     }
 
-    boolean canAccept(PendingRequest request) {
-      return queue.isEmpty()
-          || (queue.size() < streamingRpcBatchLimit
-              && (request.getBytes() + queuedBytes) < AbstractWindmillStream.RPC_STREAM_CHUNK_SIZE);
+    @Override
+    public boolean commitWorkItem(
+        String computation, WorkItemCommitRequest commitRequest, Consumer<CommitStatus> onDone) {
+      PendingRequest request = new PendingRequest(computation, commitRequest, onDone);
```

Comment:
if we can't accept the `PendingRequest` we just construct this object and throw it way.  How about passing `long request.getSerializedSize() + computation.length()` which is what `PendingRequest.getBytes()` returns into `canAccept`, and only creating the `PendingRequest` when we call add.

and only

Reply:
Done